packages/services/src/providers/perps/hyperliquid-perps.provider.ts

```diff
@@ -0,0 +1,741 @@
+        rawData: undefined,
+      };
+
+      // Sampling for raw data storage
+      const shouldSample = Math.random() < this.SAMPLING_RATE;
+
+      if (shouldSample) {
+        // Store raw data for debugging
+        summary.rawData = { clearinghouseState, recentFills };
+
+        // Also send to Sentry for monitoring
+        Sentry.captureMessage("Hyperliquid API Response Sample", {
+          level: "debug",
+          extra: {
+            response: { clearinghouseState, recentFills },
+            walletAddress: maskedAddress,
+            processingTime: Date.now() - startTime,
+          },
+        });
+
+        this.logger.debug(
+          `[HyperliquidProvider] Sampled request - storing raw data for ${maskedAddress}`,
+        );
+      }
+
+      this.logger.debug(
+        `[HyperliquidProvider] Fetched account summary for ${maskedAddress} in ${Date.now() - startTime}ms`,
+      );
+
+      return summary;
+    } catch (error) {
+      Sentry.captureException(error, {
+        extra: {
+          walletAddress: maskedAddress,
+          method: "getAccountSummary",
+          processingTime: Date.now() - startTime,
+        },
+      });
+
+      this.logger.error(
+        `[HyperliquidProvider] Error fetching account summary for ${maskedAddress}:`,
+        error,
+      );
+      throw error;
+    }
+  }
+
+  /**
+   * Get positions - transforms Hyperliquid positions to generic format
+   */
+  async getPositions(walletAddress: string): Promise<PerpsPosition[]> {
+    const startTime = Date.now();
+    const maskedAddress = this.maskWalletAddress(walletAddress);
+
+    // Add Sentry breadcrumb for debugging
+    Sentry.addBreadcrumb({
+      category: "hyperliquid.api",
+      message: `Positions request`,
+      level: "info",
+      data: {
+        walletAddress: maskedAddress,
+      },
+    });
+
+    try {
+      this.logger.debug(
+        `[HyperliquidProvider] Fetching positions for ${maskedAddress}`,
+      );
+
+      const clearinghouseState =
+        await this.getClearinghouseState(walletAddress);
+
+      // Fetch current market prices for all assets
+      const marketPrices = await this.getMarketPrices();
+
+      const positions: PerpsPosition[] = [];
+
+      for (const asset of clearinghouseState.assetPositions) {
+        const pos = asset.position;
+
+        // Skip if no position (size is 0)
+        if (!pos || pos.szi === "0") {
+          continue;
+        }
+
+        const size = new Decimal(pos.szi);
+        const isLong = size.isPositive();
+
+        // Get current price from market data or calculate from position data
+        const currentPrice = this.getCurrentPrice(pos, marketPrices);
+
+        positions.push({
+          // Identifiers (Hyperliquid doesn't provide position IDs)
+          // Use wallet + coin + size + entry price for uniqueness
+          providerPositionId: `${walletAddress}-${pos.coin}-${pos.szi}-${pos.entryPx}`,
```

Comment:
in theory, this could have a duplicate/conflict if you open the same sized position at the same price, right?

maybe use a uuid